server/vulnerabilities/oval/parsed/dpkginfo_state_evr_string.go

```diff
@@ -0,0 +1,44 @@
+package oval_parsed
+
+import (
+	"fmt"
+	"strings"
+)
+
+// This is just one possible children of <dpkginfo_state>, that said
+// all deb package tests are written against evr strings
+// see: https://oval.mitre.org/language/version5.10.1/ovaldefinition/documentation/oval-definitions-schema.html#EntityStateEVRStringType
+
+// op|value
+type ObjectStateEvrString string
+
+func NewObjectState(op string, evr string) ObjectStateEvrString {
+	return ObjectStateEvrString(fmt.Sprintf("%s|%s", op, evr))
+}
+
+func (sta ObjectStateEvrString) unpack() (OperationType, string) {
+	parts := strings.Split(string(sta), "|")
+	return NewOperationType(parts[0]), parts[1]
+}
+
+func (sta ObjectStateEvrString) Eval(ver string, cmp func(string, string) int) bool {
```

Comment:
Why does it receive a `func(string, string) int` method?

Reply:
Comparing versions is a tricky business, wanted to abstract that to make things easier to test.